src/messages/mod.rs

```diff
@@ -71,39 +93,53 @@ pub struct HopMessageWithBytes {
 
 impl HopMessageWithBytes {
     /// Serialize message and keep both SignedRoutingMessage and Bytes.
-    pub fn new(content: SignedRoutingMessage) -> Result<Self> {
-        let message = Message::Hop(content);
+    pub fn new(full_content: SignedRoutingMessage, log_ident: &LogIdent) -> Result<Self> {
+        let message = Message::Hop(full_content);
         let full_message_bytes = to_network_bytes(&message)?;
 
-        let content = if let Message::Hop(content) = message {
-            content
+        let full_content = if let Message::Hop(full_content) = message {
```

Comment:
we just composed the message, and then de-compose it?
is it better than just clone the full_content ?

Reply:
Yes, it just move parts around without cloning.